src/library/scanner/libraryscanner.cpp

```diff
@@ -251,6 +251,81 @@ void LibraryScanner::slotStartScan() {
     }
 }
 
+void LibraryScanner::cleanUpScan( const QStringList& verifiedTracks,
+        const QStringList& verifiedDirectories) {
+    // At the end of a scan, mark all tracks and directories that weren't
+    // "verified" as "deleted" (as long as the scan wasn't canceled half way
+    // through). This condition is important because our rescanning algorithm
+    // starts by marking all tracks and dirs as unverified, so a canceled scan
+    // might leave half of your library as unverified. Don't want to mark those
+    // tracks/dirs as deleted in that case) :)
+
+    // Start a transaction for all the library hashing (moved file
+    // detection) stuff.
+    ScopedTransaction transaction(m_database);
+
+    qDebug() << "Marking tracks in changed directories as verified";
+    m_trackDao.markTrackLocationsAsVerified(verifiedTracks);
+
+    qDebug() << "Marking unchanged directories and tracks as verified";
+    m_libraryHashDao.updateDirectoryStatuses(verifiedDirectories, false,
+            true);
+    m_trackDao.markTracksInDirectoriesAsVerified(verifiedDirectories);
+
+    // After verifying tracks and directories via recursive scanning of the
+    // library directories the only unverified tracks will be files that are
+    // outside of the library directories and files that have been
+    // moved/deleted/renamed.
+    qDebug() << "Checking remaining unverified tracks.";
+    if (!m_trackDao.verifyRemainingTracks(
+            m_scannerGlobal->shouldCancelPointer())) {
+        // canceled
+        return;
+    }
+
+    qDebug() << "Marking unverified tracks as deleted.";
+    m_trackDao.markUnverifiedTracksAsDeleted();
+
+    qDebug() << "Marking unverified directories as deleted.";
+    m_libraryHashDao.markUnverifiedDirectoriesAsDeleted();
+
+    // For making the scanner slow
+   qDebug() << "Burn CPU";
+   for (int i = 0;i < 1000000000; i++) asm("nop");
+
+
+
+    // Check to see if the "deleted" tracks showed up in another location,
+    // and if so, do some magic to update all our tables.
+    qDebug() << "Detecting moved files.";
```

Comment:
Why would you do that after marking things as deleted? Logically it makes for sense for me to first look for moved files and do our things. Only after all of that is done check if tracks should be marked as deleted